pkg/vsphere/disk/disk_manager.go

```diff
@@ -269,7 +272,11 @@ func (m *Manager) Get(op trace.Operation, diskURI *object.DatastorePath, fst Fil
 // }
 
 // Attach attempts to attach a virtual disk
-func (m *Manager) Attach(op trace.Operation, disk *types.VirtualDisk) error {
+func (m *Manager) Attach(op trace.Operation, config *VirtualDiskConfig) error {
+	defer trace.End(trace.Begin(""))
+
+	disk := m.toSpec(config)
```

Comment:
CreateAndAttach is the only caller, that is also why this is not taking any locks. I'm hoping to change that with a follow-up pr (planning to drop Create, provide only reconfigure based disk operations and a Wait function to make sure that disk becomes visible in the guest). 

I'll stop exporting this